sql/report_util_functions.sql

```diff
@@ -849,4 +849,101 @@ BEGIN
 
     RETURN (result);
 END$$
-DELIMITER ;
\ No newline at end of file
+DELIMITER ;
+
+-- patientHasNotBeenEnrolledIntoHivProgramDuringReportingPeriod
+
+DROP FUNCTION IF EXISTS patientHasNotBeenEnrolledIntoHivProgramDuringReportingPeriod;
+
+DELIMITER $$
+CREATE FUNCTION patientHasNotBeenEnrolledIntoHivProgramDuringReportingPeriod(
+    p_patientId INT(11),
+    p_startDate DATE,
+    p_endDate DATE) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN
+    DECLARE result TINYINT(1) DEFAULT 0;
+
+    SELECT
+        TRUE INTO result
+    FROM person p
+    JOIN patient_program pp ON pp.patient_id = p.person_id AND pp.voided = 0
+    JOIN program pro ON pro.program_id = pp.program_id AND pro.retired = 0
+    WHERE p.person_id = p_patientId
+        AND p.voided = 0
+        AND DATE(pp.date_enrolled) BETWEEN p_startDate AND p_endDate
+        AND pro.name = "HIV_PROGRAM_KEY"
+    GROUP BY pro.name;
+
+    RETURN (!result);
+END$$
+DELIMITER ;
+
+-- patientAgeIsBetween
+
+DROP FUNCTION IF EXISTS patientAgeIsBetween;  
+
+DELIMITER $$ 
+CREATE FUNCTION patientAgeIsBetween(
+    p_patientId INT(11),
+    p_startAge INT(11),
+    p_endAge INT(11),
+    p_includeEndAge TINYINT(1)) RETURNS TINYINT(1) 
+    DETERMINISTIC 
+BEGIN 
+    DECLARE result TINYINT(1) DEFAULT 0; 
+    SELECT  
+        IF (p_includeEndAge, 
+            timestampdiff(YEAR, p.birthdate, CURDATE()) BETWEEN p_startAge AND p_endAge, 
+            timestampdiff(YEAR, p.birthdate, CURDATE()) >= p_startAge
+                AND timestampdiff(YEAR, p.birthdate, CURDATE()) < p_endAge
+        ) INTO result
+        FROM person p 
+        WHERE p.person_id = p_patientId AND p.voided = 0;
+
+    RETURN (result ); 
+END$$ 
+DELIMITER ;
+
+-- patientHasPickedProphylaxisDuringReportingPeriod
+
+DROP FUNCTION IF EXISTS patientHasPickedProphylaxisDuringReportingPeriod;
+
+DELIMITER $$
+CREATE FUNCTION patientHasPickedProphylaxisDuringReportingPeriod(
+    p_patientId INT(11),
+    p_startDate DATE,
+    p_endDate DATE) RETURNS TINYINT(1)
+    DETERMINISTIC
+BEGIN
+
+    DECLARE result TINYINT(1) DEFAULT 0;
+
+    SELECT TRUE INTO result
+    FROM orders o
+    JOIN drug_order do ON do.order_id = o.order_id
+    JOIN concept c ON do.duration_units = c.concept_id AND c.retired = 0
+    JOIN drug d ON d.drug_id = do.drug_inventory_id AND d.retired = 0
+    WHERE o.patient_id = p_patientId AND o.voided = 0
+        AND drugIsProphylaxis(d.name)
+        AND o.date_activated BETWEEN p_startDate AND p_endDate
```

Comment:
@prashantsable1991 Please use `o.scheduled_date` here (it contains the date when the drug treatment started)